system-test/spanner.ts

```diff
@@ -8894,6 +8894,24 @@ describe('Spanner', () => {
         mismatchedColumnError(done, DATABASE, googleSqlTable);
       });
 
+      it('GOOGLE_STANDARD_SQL should use getTransaction for executing sql', async () => {
+        const promise = await DATABASE.getTransaction({optimisticLock: true});
+        const transaction = promise[0];
+
+        try {
+          const [rows] = await transaction!.run('SELECT * FROM TxnTable');
+          assert.strictEqual(rows.length, googleSqlRecords.length);
+        } catch (err) {
+          if ((err as grpc.ServiceError).code === grpc.status.ABORTED) {
+            assert.ok(err, 'Transaction is aborted');
```

Comment:
Pure theoretically, this could happen for other, valid reasons as well. Although optimistic locking means that Spanner won't take any exclusive locks during the query, it does not mean that Spanner will never abort the transaction for other reasons. That means that this test could still fail in a few cases.

Maybe we should add a comment here to clarify that a flaky failures on this line is acceptable, as long as the error message indicates that the error was not due to a lock conflict?

Reply:
Added the comment